tests/TestCase/ORM/Association/BelongsToManyTest.php

```diff
@@ -421,9 +416,12 @@ public function testFinderOption(): void
      */
     public function testCascadeDelete(): void
     {
-        $articleTag = $this->getMockBuilder(Table::class)
-            ->onlyMethods(['deleteAll'])
-            ->getMock();
+        $articleTag = new class (['alias' => 'Articles']) extends Table {
+            public function deleteAll($conditions): int
+            {
+                return 1;
```

Comment:
The mock asserted that the method was called with specific value for the `$conditions` arguments, that's not done anymore. Checking the a conditions set for the associations are passed to the table is important.

Reply:
Then there are other converted tests which need to be adjusted as well. I have not always added these kind of checkes since the passed down args were rarely important if the resulting object was correct imho.

Reply:
... since the passed down args were rarely important if the resulting object was correct imho.

Such an assumptions would hold true only if the arguments are actually used within the method. In this particular case you are simply ignoring the arguments and returning `1`.